core/state_transition.go

```diff
@@ -391,6 +400,17 @@ func (st *StateTransition) innerTransitionDb() (*ExecutionResult, error) {
 	}
 	st.state.AddBalance(st.evm.Context.Coinbase, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), effectiveTip))
 
+	if st.evm.ChainConfig().Optimism != nil {
+		optimismConfig := st.evm.ChainConfig().Optimism
+		if optimismConfig.Enabled {
+			st.state.AddBalance(optimismConfig.FeeRecipient, new(big.Int).Mul(new(big.Int).SetUint64(st.gasUsed()), st.evm.Context.BaseFee))
+			// TODO: handle L1Cost being nil
```

Comment:
Probably revert Tx. Only issue is that deposits should not have an L1 Cost.

Reply:
OK, we have a condition in `AsMessage` where we can override and always set `l1Cost = nil` for Deposit transactions.

Reply:
Adding an assertion in `ApplyTransaction` make sure other transactions always include `l1Cost`.